### PR TITLE
Feature/fix server responses bad captures

### LINKED ISSUE #2274: Conan client say "Binary for xxxx not in remote:" when it's a 500 error
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
1.0.0-beta5
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

https://travis-ci.org/lasote/conan-openssl/jobs/326398470

It's trying to download zlib and failing with a 500 error, but Conan instead of fail, it's considering that the package is missing.

### PR DESCRIPTION
**Pull request by lasote**
- [x] Refer to the issue that supports this Pull Request. #2274
- [x] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.

closes #2274

### CODE REVIEW

File: conans/client/remote_registry.py [hunk 1]
```diff
@@ -98,8 +98,8 @@ def remote(self, name):
             try:
                 return Remote(name, remotes[name][0], remotes[name][1])
             except KeyError:
-                raise ConanException("No remote '%s' defined in remotes in file %s"
-                                     % (name, self._filename))
+                raise NoRemoteAvailable("No remote '%s' defined in remotes in file %s"
+                                        % (name, self._filename))
```
Comment (pvicente) [inline conans/client/remote_registry.py]: leading whitespace?

Comment (memsharded) [reply conans/client/remote_registry.py]: Nop, just aligning text for pep8, isn't?

Comment (pvicente) [reply conans/client/remote_registry.py]: yes a bit misleading view in Github, sorry

File: conans/test/download_test.py [hunk 2]
```diff
@@ -22,6 +24,59 @@ class HelloConan(ConanFile):
...
+
+            def __init__(self, ok, status_code):
+                self.ok = ok
+                self.status_code = status_code
+
+        class BuggyRequester(object):
+            def get(self, *args, **kwargs):
+                return Response(False, 404)
+
+        client2.remote_manager._remote_client._rest_client.requester = BuggyRequester()
```
Comment (pvicente) [inline conans/test/download_test.py]: Not a fan about it, going to submit a pull request and check what do you think about my changes, better than explaining it here.

See [pull request](https://github.com/lasote/conan/pull/12)

CC: @memsharded

**Comment by lasote** [comment]
I implemented an intermediate solution of the test, without mock classes but injecting the dependency to the test client.

**Pull request merged**